streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -161,13 +161,15 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
         if (retentionPeriod < 0) {
             throw new IllegalArgumentException("retentionPeriod cannot be negative");
         }
-        if (numSegments < 1) {
-            throw new IllegalArgumentException("numSegments cannot must smaller than 1");
+        if (numSegments < 2) {
+            throw new IllegalArgumentException("numSegments cannot must smaller than 2");
```

Comment:
Previously, the `RocksDbWindowBytesStoreSupplier` constructor enforced the minimum of 2, so if you passed `< 1`, you'd get an exception from here, and if you passed `<2`, you'd get the exception from the constructor called just below here.

This way, we collapse both checks into one equivalent check here.